cdap-app-templates/cdap-etl/hydrator-test/src/main/java/co/cask/cdap/etl/mock/spark/streaming/MockSink.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.etl.mock.spark.streaming;
+
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.dataset.DatasetProperties;
+import co.cask.cdap.api.dataset.lib.KeyValueTable;
+import co.cask.cdap.api.plugin.PluginClass;
+import co.cask.cdap.api.plugin.PluginConfig;
+import co.cask.cdap.api.plugin.PluginPropertyField;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.batch.SparkExecutionPluginContext;
+import co.cask.cdap.etl.api.batch.SparkPluginContext;
+import co.cask.cdap.etl.api.batch.SparkSink;
+import co.cask.cdap.etl.proto.v2.ETLPlugin;
+import co.cask.cdap.test.DataSetManager;
+import com.google.gson.Gson;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.function.Function;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Mock Sink for Spark streaming test.
+ */
+public class MockSink extends SparkSink<StructuredRecord> {
+  public static final PluginClass PLUGIN_CLASS = getPluginClass();
+  private static final Gson GSON = new Gson();
+  private final Config config;
+
+  public MockSink(Config config) {
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    super.configurePipeline(pipelineConfigurer);
+    pipelineConfigurer.createDataset(config.tableName, KeyValueTable.class.getName(), DatasetProperties.EMPTY);
+  }
+
+  @Override
+  public void prepareRun(SparkPluginContext context) throws Exception {
+  }
+
+  @Override
+  public void run(final SparkExecutionPluginContext context, JavaRDD<StructuredRecord> input) throws Exception {
+    List<Integer> ids = input.map(new Function<StructuredRecord, Integer>() {
+      @Override
+      public Integer call(StructuredRecord input) throws Exception {
+        String id = input.get("id");
+        return Integer.parseInt(id);
+      }
+    }).collect();
```

Comment:
this isn't how we would recommend people to write to a table. It should use context.saveAsDataset(). Should basically avoid collect() if possible.